redis/redlock.go

```diff
@@ -2,31 +2,46 @@ package redis
 
 import (
 	"context"
+	"fmt"
 	"time"
 
 	"github.com/ElrondNetwork/elrond-go-core/core/check"
+	"github.com/ElrondNetwork/notifier-go/common"
 )
 
-const expiry = time.Minute * 30
+const minTTLInMinutes = 0
+
+type ArgsRedlockWrapper struct {
+	Client       RedLockClient
+	TTLInMinutes uint32
+}
 
 type redlockWrapper struct {
 	client RedLockClient
+	ttl    time.Duration
 }
 
 // NewRedlockWrapper create a new redLock based on a cache instance
-func NewRedlockWrapper(client RedLockClient) (*redlockWrapper, error) {
-	if check.IfNil(client) {
+func NewRedlockWrapper(args ArgsRedlockWrapper) (*redlockWrapper, error) {
+	if check.IfNil(args.Client) {
 		return nil, ErrNilRedlockClient
 	}
+	if args.TTLInMinutes <= minTTLInMinutes {
```

Comment:
Since `TTLInMinutes` is `uint32` you could never have a negative value here. I think it is enough to just check for zero value and return `ErrZeroValueReceived`

Reply:
:+1: changed to check for zero value